packages/opentelemetry-core/src/platform/node/id.ts

```diff
@@ -14,23 +14,27 @@
  * limitations under the License.
  */
 
-import * as crypto from 'crypto';
-
 const SPAN_ID_BYTES = 8;
 const TRACE_ID_BYTES = 16;
 
 /**
  * Returns a random 16-byte trace ID formatted/encoded as a 32 lowercase hex
  * characters corresponding to 128 bits.
  */
-export function randomTraceId(): string {
-  return crypto.randomBytes(TRACE_ID_BYTES).toString('hex');
-}
+export const randomTraceId = getIdGenerator(TRACE_ID_BYTES);
 
 /**
  * Returns a random 8-byte span ID formatted/encoded as a 16 lowercase hex
  * characters corresponding to 64 bits.
  */
-export function randomSpanId(): string {
-  return crypto.randomBytes(SPAN_ID_BYTES).toString('hex');
+export const randomSpanId = getIdGenerator(SPAN_ID_BYTES);
+
+const SHARED_BUFFER = Buffer.allocUnsafe(TRACE_ID_BYTES);
+function getIdGenerator(bytes: number): () => string {
+  return function generateId() {
+    for (let i = 0; i < bytes; i++) {
+      SHARED_BUFFER[i] = Math.floor(Math.random() * 256);
+    }
+    return SHARED_BUFFER.slice(0, bytes).toString('hex');
```

Comment:
I think `SHARED_BUFFER.toString('hex', 0, bytes)` would avoid creation of a temporary `Buffer` with same result.

Reply:
Looks like this improved time by ~20%

```
  crypto                x   427,138 ops/sec ±0.77% (92 runs sampled)
  pseudorandom          x 2,767,795 ops/sec ±1.00% (93 runs sampled)
  pseudorandom no alloc x 3,355,721 ops/sec ±0.96% (90 runs sampled)
```

Reply:
there is one more thing here to be improved (about 8%)
this line
```
SHARED_BUFFER[i] = Math.floor(Math.random() * 256);
```
replace with
```
SHARED_BUFFER[i] = Math.random() * 256 | 0;
```

Reply:
33e9a4a